cumulusci/tasks/robotframework/robotdoc.py

```diff
@@ -0,0 +1,98 @@
+import jinja2
+import robot.utils
+
+import cumulusci
+from cumulusci.core.exceptions import TaskOptionsError
+from cumulusci.core.tasks import BaseTask
+from cumulusci.core.utils import process_list_arg
+
+from robot.libdocpkg import LibraryDocumentation
+from robot.libraries.BuiltIn import RobotNotRunningError
+
+
+class RobotDoc(BaseTask):
+    task_options = {
+        "outputdir": {
+            "description": "The path to a folder where generated files will be placed",
+            "required": True,
+        },
+        "files": {
+            "description": "A list of keyword definition files (eg: .py, .robot, .resource)",
+            "required": True,
+        },
+    }
+
+    def _validate_options(self):
+        super(RobotDoc, self)._validate_options()
+
+        self.options["files"] = process_list_arg(self.options["files"])
+        if not os.path.exists(self.options["outputdir"]):
+            os.mkdir(self.options["outputdir"])
+
+        bad_files = []
+        for input_file in self.options["files"]:
+            if not os.path.exists(input_file):
+                bad_files.append(input_file)
+
+        if bad_files:
+            if len(bad_files) == 1:
+                error_message = "Unable to find the input file '{}'".format(
+                    bad_files[0]
+                )
+            else:
+                files = ", ".join(["'{}'".format(filename) for filename in bad_files])
+                error_message = "Unable to find the following input files: {}".format(
+                    files
+                )
+            raise TaskOptionsError(error_message)
+
+    def _run_task(self):
+        output_dir = self.options["outputdir"]
+        libraries = []
+        for input_file in sorted(self.options["files"]):
+            basename = os.path.basename(input_file)
+            try:
+                libdoc = LibraryDocumentation(input_file)
+                libraries.append(libdoc)
+                # robot doesn't save the orginal name but we want to use that
+                # in our generated file
+                libdoc.basename = basename
+
+                # if we want to save the official libdoc file, uncomment the following
+                # two lines:
+                # libdoc.save(self.options['output_file'], "HTML")
+                # self.logger.info("created {}".format(output_file))
+
+            except RobotNotRunningError as e:
+                # oddly, robot's exception has a traceback embedded in the message, so we'll
+                # only print out the first line to hide most of the noise
+                self.logger.warn("unexpected error: {}".format(str(e).split("\n")[0]))
+
+        stylesheet_path = os.path.join(os.path.dirname(__file__), "stylesheet.css")
+        with open(stylesheet_path) as f:
+            stylesheet = f.read()
+
+        with open(os.path.join(output_dir, "index.html"), "w") as f:
+            title = "{} Keywords".format(self.project_config.project__package__name)
```

Comment:
We generally try to avoid direct references to the project_config in task code because it keeps the tasks less coupled to the rest of CCI and easier to test.

Instead I would recommend:
1. Add a "title" option to the task.
2. In cumulusci.yml where the task is defined, set the title option to `$project_config.project__package__name`

Reply:
ok. Thanks for the suggestion. I wasn't sure if what I was doing was kosher or not.

Reply:
I've made this change, though it means the default will be something like "CumulusCI" rather  than "CumulusCI Keywords", but that's not a big deal.